OpenRA.Mods.Common/Traits/Render/WithInfantryBody.cs

```diff
@@ -25,9 +25,21 @@ public class WithInfantryBodyInfo : UpgradableTraitInfo, IRenderActorPreviewSpri
 
 		[SequenceReference] public readonly string MoveSequence = "run";
 		[SequenceReference] public readonly string AttackSequence = null;
+
+		[FieldLoader.LoadUsing("LoadWeaponSequences")]
+		[SequenceReference] public readonly Dictionary<string, string> AttackSequenceByArmament = new Dictionary<string, string>();
```

Comment:
The name is a bit overboard IMO, `AttackSequences` with a desc explaining that it is a dictionary of armament names to animation would be fine.